src/lib/dma-trace.c

```diff
@@ -391,9 +391,29 @@ void dtrace_event(const char *e, uint32_t length)
 
 void dtrace_event_atomic(const char *e, uint32_t length)
 {
+	struct dma_trace_buf *buffer = NULL;
```

Comment:
This looks exactly like dtrace_event() now without the locking. The purpose of the atomic call is for low frequency and fast trace messages. If we are overflowing then the atomic user is sending too many trace messages and should be converted to verbose_trace

Reply:
linked to https://github.com/thesofproject/sof/issues/17